editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/bean/AttributeTextParameterPropertyModifyListenerForAdvanced.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.properties.bean;
+
+import org.fusesource.ide.camel.editor.properties.creators.modifylisteners.text.AbstractTextParameterPropertyModifyListener;
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+import org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+/**
+ * @author brianf
+ *
+ */
+public class AttributeTextParameterPropertyModifyListenerForAdvanced extends AbstractTextParameterPropertyModifyListener {
+
+	private Parameter parameter;
+
+	public AttributeTextParameterPropertyModifyListenerForAdvanced(AbstractCamelModelElement camelModelElement, Parameter parameter) {
+		super(camelModelElement, parameter.getName());
+		this.parameter = parameter;
+	}
+
+	@Override
+	protected void updateModel(String newValue) {
+		setAttributeValue(this.parameter.getName(), newValue);
+	}
+	
+	private void setAttributeValue(String attrName, String attrValue) {
```

Comment:
this method is working directly with the xml node, it would be better to delegate that to the model (so I suppose here to move the code to the CamelBean class.)
it will allow:
- to ensure that the model is in sync
- maybe to write a test manipulating the model